BatteryDataStreamer/BatteryDataStreamer/BatteryInputGenerator.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BatteryDataStreamer
+{
+    class BatteryInputGenerator: IBatteryInputGenerator
+    {
+         private readonly Random randomNumberGenerator = new Random();
+        public BmsBattery GetBatteryInputParameters()
+        {
+            BmsBattery BatteryProperties = new BmsBattery();
+            BatteryProperties.Temperature = GetBatteryTemperature();
+            BatteryProperties.ChargeRate = GetBatteryChargeRate();
+            return BatteryProperties;
+        }
+        public double GetBatteryTemperature()
```

Comment:
`Get...` functions are normally expected to return the latest value in a single operation. However, these get-functions are also delaying, maybe to simulate actual fetch from the device.